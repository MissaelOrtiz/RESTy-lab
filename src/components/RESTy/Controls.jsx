/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../styles/RestyCss.css';

const Controls = ({ valueUrl, valueRawJson, onChange, onSubmit }) => (
  <div className={styles.controls}>
    <form onSubmit={onSubmit} onChange={onChange}>
      <input type="text" aria-label="urlTextbox" placeholder="URL" value={valueUrl} name="url" className={styles.urlTextbox}/>
      <section aria-label="methodSelector" className={styles.methodSelector}>
        <label htmlFor="GET">
          <input type="radio" name="method" value="GET" id="GET"/>
          <div className={styles.selectorName}>GET</div>
        </label>
        <label htmlFor="POST">
          <input type="radio" name="method" value="POST" id="POST"/>
          <div className={styles.selectorName}>POST</div>
        </label>
        <label htmlFor="PUT">
          <input type="radio" name="method" value="PUT" id="PUT"/>
          <div className={styles.selectorName}>PUT</div>
        </label>
        <label htmlFor="PATCH">
          <input type="radio" name="method" value="PATCH" id="PATCH"/>
          <div className={styles.selectorName}>PATCH</div>
        </label>
        <label htmlFor="DELETE">
          <input type="radio" name="method" value="DELETE" id="DELETE"/>
          <div className={styles.selectorName}>DELETE</div>
        </label>
        <button className={styles.selectorName}>GO!</button>
      </section>
      <input type="text" aria-label="rawJsonTextbox" placeholder="Raw JSON body" value={valueRawJson} name="rawJson" className={styles.rawJsonTextbox}/>
    </form>
  </div>
);

Controls.propTypes = {
  valueUrl: PropTypes.string.isRequired,
  valueRawJson: PropTypes.string.isRequired, 
  onChange: PropTypes.func.isRequired, 
  onSubmit: PropTypes.func.isRequired
};

export default Controls;
