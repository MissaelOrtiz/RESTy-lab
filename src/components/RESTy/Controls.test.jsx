/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Controls from './Controls';

describe('Controls', () => {
  it('displays a form with two inputs, five radio selections, and  a button', () => {
    render(
      <Controls 
        valueUrl=""
        valueRawJson=""
        onChange={() => {}}
        onSubmit={() => {}}
      />
    );

    const controlsForm = screen.getByTestId('controls-form');
    expect(controlsForm).toMatchSnapshot();
  });
});
