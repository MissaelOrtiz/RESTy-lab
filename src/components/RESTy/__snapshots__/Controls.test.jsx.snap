// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls displays a form with two inputs, five radio selections, and  a button 1`] = `
<form
  data-testid="controls-form"
>
  <input
    aria-label="urlTextbox"
    class="urlTextbox"
    name="url"
    placeholder="URL"
    type="text"
    value=""
  />
  <section
    aria-label="methodSelector"
    class="methodSelector"
  >
    <label
      for="GET"
    >
      <input
        id="GET"
        name="method"
        type="radio"
        value="GET"
      />
      <div
        class="selectorName"
      >
        GET
      </div>
    </label>
    <label
      for="POST"
    >
      <input
        id="POST"
        name="method"
        type="radio"
        value="POST"
      />
      <div
        class="selectorName"
      >
        POST
      </div>
    </label>
    <label
      for="PUT"
    >
      <input
        id="PUT"
        name="method"
        type="radio"
        value="PUT"
      />
      <div
        class="selectorName"
      >
        PUT
      </div>
    </label>
    <label
      for="PATCH"
    >
      <input
        id="PATCH"
        name="method"
        type="radio"
        value="PATCH"
      />
      <div
        class="selectorName"
      >
        PATCH
      </div>
    </label>
    <label
      for="DELETE"
    >
      <input
        id="DELETE"
        name="method"
        type="radio"
        value="DELETE"
      />
      <div
        class="selectorName"
      >
        DELETE
      </div>
    </label>
    <button
      class="selectorName"
    >
      GO!
    </button>
  </section>
  <input
    aria-label="rawJsonTextbox"
    class="rawJsonTextbox"
    name="rawJson"
    placeholder="Raw JSON body"
    type="text"
    value=""
  />
</form>
`;
